### ISSUE TITLE
new_version_cp should not require a connection to the database

### ISSUE THREAD

**Issue by esabol** [bucardo/check_postgres#136]
I think this is a regression from previous versions.

```shell
> ./check_postgres --action=new_version_cp
ERROR: could not connect to server: No such file or directory
        Is the server running locally and accepting
        connections on Unix domain socket "/tmp/.s.PGSQL.5432"?
```
The connection to the database is initiated by the following line in `find_new_version()`:
```Perl
    my $version = verify_version();
```
Why was this added? It's not in older versions. If I comment it out, it works as expected....

**Comment by machack666** [comment]
Hi Ed,

Ah, this is because the version-checking code uses the same underlying routine for postgresql as the other versions, and the version-numbering structure for postgresql changed with the move to Pg 10.  I'll review and see what might be an adequate solution here that doesn't assume a local PostgreSQL connection.

**Comment by esabol** [comment]
Hmmm, I don't really see how that works in practice. `find_new_version()` is only called by four routines:

* `check_new_version_bc()`
* `check_new_version_box()`
* `check_new_version_cp()`
* `check_new_version_tnm()`
* `check_new_version_pg()`

I think the only one of those that needs to connect to a database is the last one, and it already does so _prior_ to calling `find_new_version()` with the call `run_command('SELECT version() AS version')`. So I'm not sure what the call to `verify_version()` in `find_new_version()` is accomplishing. The return value stored in `$version` is not subsequently used or referenced in `find_new_version()`. If I comment it out, `new_version_cp` and `new_version_pg` seemingly still work as intended. I can't seem to get `new_version_box` or `new_version_bc` or `new_version_tnm` to work, regardless.

**Comment by machack666** [comment]
@esabol yeah, it may well be some residual junk in there.  What version did you test against; I'm mainly wanting to verify that 10.1 still works without issue.  Will review and verify that commenting it out will fix it.

**Comment by esabol** [comment]
I have tested with both 10.1 and 9.2.24.

**Comment by machack666** [comment]
Okay, also verified that things are sane without, and pointless with.  Thanks for the report; fixed in d9505e4.

**Comment by esabol** [comment]
Thanks for the quick resolution!
